### DISCUSSION TITLE
Disable clearing selection on ESC key for <ListBox />

### DISCUSSION THREAD

**Discussion by apucacao** [adobe/react-spectrum#5678] [General]
Hello,

Is there a way to disable clearing listbox selection when pressing the Escape key?

As far as I can tell, the only way to do that right now would be to drop down to `useListBox` (https://github.com/adobe/react-spectrum/discussions/2268). Which is unfortunate because that would mean losing all the benefits of `<ListBox />`.

Here's what we're trying to do: build a GitHub-like PR reviewer dialog. The dialog is composed of the folllowing:
- `Button`: this is the dialog trigger
- `Popover`
- `Dialog`
- `SearchField`: allows filtering the items via our API
- `ListBox`: for selection

The UX is almost there: users can toggle the popover dialog via the button, filter items, etc… They can then tab over to the listbox and make their selection.

Then they can press ESC to close the popover. But that ends up clearing the list box as well.

I wonder if I'm overlooking something.

**Comment by reidbarber** [comment]
There was some discussion on this in https://github.com/adobe/react-spectrum/issues/4476 as well. I'm not sure that we decided this is a good idea since there wouldn't be another way to clear the ListBox selection via keyboard. I think this is probably still up for debate.

**Reply by apucacao** [reply]
Yeah, thinking about it some more, I'm beginning to see ESC-to-clear as useful for keyboard users.

One downside is that I'm not sure it's useful to simultaneously close the popover. 🤔

**Reply by snowystinger** [reply]
Agreed, it should result in only one action. I think the answer was in @devongovett's comment from the other linked issue
> Basically it should stop propagation when escape is pressed if there is a selection, and otherwise allow propagation to the overlay.

**Reply by apucacao** [reply]
Thanks, I missed that.

If I understand correctly, I can't ensure that happens ^ that way if I'm using `<ListBox />` at the moment, correct? As far as I can tell, I don't have access to the keyboard event.

**Reply by snowystinger** [reply]
Correct, that is the fix we'd need to make internally. We also accept PRs :)
